app/models/queries/filters/strategies/list_contains.rb

```diff
@@ -0,0 +1,34 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Queries::Filters::Strategies
+  class ListContains < List
+    self.supported_operators = ["~"]
+    self.default_operator = "~"
+  end
+end
```

Comment:
Oddly enough, this did not exist or I'm missing some context- so please shout if I should be approaching this differently.

Reply:
If you use the `List` strategy, it will have to operators: `=` and `!`. In the context of the list, the `=` operator maps to [Queries::Operators::EqualsOr which is actually Queries::Operators::Equals](https://github.com/opf/openproject/blob/dev/app/models/queries/operators/equals.rb#L45-L46). That operator already matches on a subset which I think is what you are trying to achieve.

That is also present in the frontend where there is a distinction between "is (or)" and "is (and)":
<img width="1482" alt="image" src="https://github.com/opf/openproject/assets/617519/728b65fb-a7be-40f0-bcc4-57b09bc79bed">

Reply:
I stumbled over the exact same thing. It is true, what Jens says, we are using `=` as a `contains` operator on lists. You could understand it as the values are an array of values that can match, concatenated with `OR`. So, a project matches the filter, if it has ANY of the project custom fields assigned, that where passed into the filter as an array.

🔴 -> the filter type should be standard `:list`

There is on difference to e.g. the `user_action_filter`, where a project must match ALL the values given to the filter, so basically an `is (AND)`. This would be filter type `:list_all`.

Reply:
Update: I do understand that correctly, yes? This filter is supposed to be `is (OR)`?